examples/baroclinic_adjustment_with_bgc.jl

```diff
@@ -0,0 +1,280 @@
+# # Baroclinic adjustment
+#
+# In this example, we simulate the evolution and equilibration of a baroclinically
+# unstable front.
+#
+# ## Install dependencies
+#
+# First let's make sure we have all required packages installed.
+
+# ```julia
+# using Pkg
+# pkg"add Oceananigans, CairoMakie"
+# ```
+
+using Oceananigans
+using Oceananigans.Units
+using ClimaOceanBiogeochemistry: NutrientsPlanktonBacteriaDetritus
+
+# ## Grid
+
+# We use a three-dimensional channel that is periodic in the `x` direction:
+
+Lx = 1000kilometers # east-west extent [m]
+Ly = 1000kilometers # north-south extent [m]
+Lz = 200            # depth [m]
+
+Nx = 64
+Ny = 64
+Nz = 40
+
+grid = RectilinearGrid(size = (Nx, Ny, Nz),
+                       x = (0, Lx),
+                       y = (-Ly/2, Ly/2),
+                       z = (-Lz, 0),
+                       topology = (Periodic, Bounded, Bounded))
+
+# ## Model
+
+# We built a `HydrostaticFreeSurfaceModel` with an `ImplicitFreeSurface` solver.
+# Regarding Coriolis, we use a beta-plane centered at 45° South.
```

Comment:
some comment here would be appropriate e.g. in the spirit of

"and we add a biogeochemistry model for this and that."

Reply:
Appreciate the feedback - this example is going to be developed further in another P/R (see #15 and #16)

Reply:
Gotcha!